### PR TITLE
Add code and documentation to monitor scipy optimization more easily.

### PR DESCRIPTION
**Pull request by vatsalaggarwal**
**PR type:** enhancement + doc improvement

**Related issue(s)/PRs:** 

## Summary

**Proposed changes**

* ... Adds a ScipyMonitor class. An instance of this class with the step_callback argument functionality of optimizers.Scipy().minimize to monitor trainings of GP with this optimizer.
* ... The monitoring notebook is updated with a comment informing the reader what to do if they want to monitor a training with the optimizers.Scipy optimizer.

**What alternatives have you considered?**

- Adding another function to the Monitor class that returns the callable function that can be passed to step_callback. Instead, we went for adding another class as it seemed less dirty.
- For the notebook, an alternative is to add the scipy optimization + monitoring code to a runnable cell, but this breaks the flow of the notebook, so instead we chose to leave it as a comment.

### Minimal working example

```python
# Put your example code in here
```

## PR checklist

- [X] New features: code is well-documented
  - [X] detailed docstrings (API documentation)
  - [X] notebook examples (usage demonstration)
- [ ] The bug case / new feature is covered by unit tests
- [X] Code has type annotations
- [X] I ran the black formatter (`make format`)
- [X] I locally tested that the tests pass (`make check-all`)

### Release notes

**Fully backwards compatible:** yes / no

**If not, why is it worth breaking backwards compatibility:**

**Commit message (for release notes):**
Add code and documentation to monitor scipy optimization more easily.

### CODE REVIEW

**Review by vdutor — APPROVED** [review]
Thanks for the PR. LGTM.

File: gpflow/monitor/base.py [hunk 1]
```diff
@@ -18,7 +18,6 @@
 
 import tensorflow as tf
 
-
 __all__ = ["MonitorTask", "ExecuteCallback", "MonitorTaskGroup", "Monitor"]
```
Comment (vdutor) [inline gpflow/monitor/base.py:21]:
```suggestion
__all__ = ["MonitorTask", "ExecuteCallback", "MonitorTaskGroup", "Monitor", "ScipyMonitor"]
```

**Comment by st--** [comment]
How about, instead of a new ScipyMonitor class, just adding a `scipy_callback` method or something like that that has the right function signature ?

File: doc/source/notebooks/basics/monitoring.pct.py [hunk 2]
```diff
@@ -128,6 +128,21 @@ def plot_prediction(fig, ax):
...
+# something slightly different.
+#
+# ```
+# # define fast_tasks, slow_tasks, training_loss as in done in the previous 2 cells. Then,
+# from gpflow.monitor import ScipyMonitor
+# monitor = ScipyMonitor(fast_tasks, slow_tasks)
+#
+# opt = gpflow.optimizers.Scipy()
+# opt.minimize(training_loss, model.trainable_variables, step_callback=monitor)
+# ```
```
Comment (st--) [inline doc/source/notebooks/basics/monitoring.pct.py:144]: Can you turn this into an actual code cell ? That would ensure the code is actually run in the continuous integration and give us more confidence it is (and will stay) correct.

**Comment by st--** [comment]
This PR should resolve #1449

**Pull request closed without merging**